src/mvt_sql.py

```diff
@@ -0,0 +1,455 @@
+                % (column_name, column_name, params[1]),
+            )
+        if t == ">":
+            return (
+                params[0],
+                """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) > %s ELSE false END) """
+                % (column_name, column_name, params[1]),
+            )
+        if t == ">=":
+            return (
+                params[0],
+                """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) >= %s ELSE false END) """
+                % (column_name, column_name, params[1]),
+            )
+    except KeyError:
+        pass
+
+
+def get_sql_hints(choosers, obj, zoom):
+    needed = set(
+        [
+            "width",
+            "fill-color",
+            "fill-image",
+            "icon-image",
+            "text",
+            "extrude",
+            "background-image",
+            "background-color",
+            "pattern-image",
+            "shield-text",
+        ]
+    )
+
+    hints = []
+    for chooser in choosers:
+        tags = set()
+        qs = []
+        if not needed.isdisjoint(set(chooser.styles[0].keys())):
+            for rule in chooser.ruleChains:
+                if obj:
+                    if (rule.subject != "") and not _test_feature_compatibility(
+                        obj, rule.subject
+                    ):
+                        continue
+                if not rule.test_zoom(zoom):
+                    continue
+                b = set()
+                for condition in rule.conditions:
+                    q = get_sql(condition, obj)
+                    if q:
+                        if q[1]:
+                            tags.add(q[0])
+                            b.add(q[1])
+
+                if b:
+                    qs.append("(" + " AND ".join(b) + ")")
+
+        if qs:
+            hints.append((tags, " OR ".join(qs)))
+
+    return hints
+
+
+def get_vectors(minzoom, maxzoom, x, y, style, vec):
+    geomcolumn = "way"
+
+    pxtolerance = 0.5
+    table = {
+        "polygon": "planet_osm_polygon",
+        "line": "planet_osm_line",
+        "point": "planet_osm_point",
+    }
+
+    types = {"line": "line", "polygon": "area", "point": "node"}
+
+    column_names = set()
+    adp = ""
+
+    adp = []
+
+    for zoom in range(minzoom, maxzoom + 1):
+        column_names.update(style.get_interesting_tags(types[vec], zoom))
+
+        sql_hint = get_sql_hints(style.choosers, types[vec], zoom)
+        for tp in sql_hint:
+            add = []
+            for j in tp[0]:
+                if j not in column_names:
+                    break
+            else:
+                add.append(tp[1])
+            if add:
+                add = " OR ".join(add)
+                add = "(" + add + ")"
+                adp.append(add)
+
+    if "name:en" in column_names:
```

Comment:
It will not be in column names, it will not be mentioned in styles. Should probably just always be here, or passed as list of languages to support from user.